examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp

```diff
@@ -172,35 +213,37 @@ EmberAfStatus OTAProviderExample::HandleQueryImage(chip::app::CommandHandler * c
 
     switch (mQueryImageBehavior)
     {
-    case kRespondWithUnknown:
+    case kRespondWithUnknown: {
         // This use-case is a subset of the ota-candidates-file option.
         // Can be removed once all other platforms start using the ota-candidates-file method.
         if (strlen(mOTAFilePath) > 0) // If OTA file is directly provided
         {
-            // TODO: Following details shall be read from the OTA file
-
-            // If software version is provided using command line then use it
-            if (mSoftwareVersion.HasValue())
-            {
-                newSoftwareVersion = mSoftwareVersion.Value();
-            }
-
-            // If software version string is provided using command line then use it
-            if (mSoftwareVersionString)
-            {
-                newSoftwareVersionString = mSoftwareVersionString;
-            }
+            // Parse the header and set version info based on the header
+            VerifyOrDie(ParseOTAHeader(mOTAFilePath, header) == true);
+            VerifyOrDie(SW_VER_STR_MAX_LEN > header.mSoftwareVersionString.size());
```

Comment:
```suggestion
            VerifyOrDie(sizeof(newSoftwareVersionString) > header.mSoftwareVersionString.size());
```
because that's what the code below this really relies on.